executor/memtable_reader.go

```diff
@@ -531,10 +539,19 @@ func (e *clusterLogRetriever) startRetrieving(ctx context.Context, sctx sessionc
 						return
 					}
 					if err != nil {
-						ch <- logStreamResult{addr: address, typ: serverType, err: err}
+						select {
+						case ch <- logStreamResult{addr: address, typ: serverType, err: err}:
+						case <-ctx.Done():
```

Comment:
Maybe `ctx.Done()` should be front of `ch <- logStreamResult`, to make `ctx.Done()` the first choice when they're both ready?

Reply:
It's doesn't matter here because the `ch` has no buffer.